File: 2019/j4.py
# code by savir singh

orientations = input()
grid =[[1,2], [3,4]]
# 1 2
# 3 4

for i in range(len(orientations)):
    if orientations[i] == "H":
        grid[1], grid[0] = grid[0], grid[1]

    elif orientations[i] == "V":
        grid[0][0], grid[0][1] = grid[0][1], grid[0][0]
        grid[1][0], grid[1][1] = grid[1][1], grid[1][0]

print(grid[0][0], grid[0][1])
print(grid[1][0], grid[1][1])

File: 2019/j4.cpp
// code by savir singh

#include <bits/stdc++.h>
#define ll long long
#define del erase
using namespace std;

int main() {
    vector<vector<ll>> original = {{1, 2}, {3, 4}};
    string input;
    cin >> input;
    for (int i=0; i<input.size(); i++) {
        // horizontal flip
        if (input[i]=='H') {
            vector<ll> wait=original[0];
            original[0]=original[1];
            original[1]=wait;
        }
        // vertical flip
        else {
            ll wait=original[0][0];
            original[0][0] = original[0][1];
            original[0][1] = wait;
            wait=original[1][0];
            original[1][0] = original[1][1];
            original[1][1] = wait;
        }
    }
    for (int j=0; j<original.size(); j++) {
        cout << original[j][0] << " " << original[j][1] << "\n";
    }
}
